modules/core/src/minmax.cpp

```diff
@@ -1510,18 +1510,29 @@ void cv::minMaxIdx(InputArray _src, double* minVal,
 
     Mat src = _src.getMat(), mask = _mask.getMat();
 
-    int _minIdx, _maxIdx;
-    int* min_offset = (cn == 1) ? minIdx : &_minIdx;
-    int* max_offset = (cn == 1) ? maxIdx : &_maxIdx;
     if (src.dims <= 2)
     {
-        CALL_HAL(minMaxIdx, cv_hal_minMaxIdx, src.data, src.step, src.cols*cn, src.rows, src.depth(),
-                    minVal, maxVal, min_offset, max_offset, mask.data);
+        CALL_HAL(minMaxIdx, cv_hal_minMaxIdx, src.data, src.step, src.cols*cn, src.rows,
+                 src.depth(), minVal, maxVal, minIdx, maxIdx, mask.data);
     }
     else if (src.isContinuous())
     {
-        CALL_HAL(minMaxIdx, cv_hal_minMaxIdx, src.data, 0, (int)src.total()*cn, 1, src.depth(),
-                 minVal, maxVal, min_offset, max_offset, mask.data);
+        int res = cv_hal_minMaxIdx(src.data, 0, (int)src.total()*cn, 1, src.depth(),
+                                   minVal, maxVal, minIdx, maxIdx, mask.data);
+
+        if (res == CV_HAL_ERROR_OK)
+        {
+            if (minIdx)
+                ofs2idx(src, minIdx[0], minIdx);
```

Comment:
@mplatings It's not clear from the docs if `minIdx` and `maxIdx` should contain pointers to an offset to convert it to array indices or to indices itself.
In the first case (which how OpenCV tab functions from `getMinmaxTab()` actually work) there should be 0th index and no such thing as `minIdx[i]` or `maxIdx[i]` is ever possible for every i > 0.
In the second case you're right since HAL call gets 1-dimensional array and puts Y index to `minIdx[0]` (which is always 0) and X to `minIdx[1]` (the same for `maxIdx`).
We won't solve the problem until we clarify this somehow.

By the way, can you tell me what tests are failing? Looks like OpenCV does not cover that case.

Reply:
KleidiCV's tests fail with the head of the OpenCV 4.x branch. Specifically this: https://gitlab.arm.com/kleidi/kleidicv/-/blob/main/conformity/opencv/test_min_max.cpp?ref_type=heads#L59

In KleidiCV 0.1.0 we did not enable minmax in the HAL so from our point of view it's fine for us to change how the HAL should work. If the HAL code should return an offset instead of 2-dimensional indices then that makes KleidiCV's code simpler.

So I'm happy to either change these indices from 0 to 1, or to require that the HAL returns an offset and we will change KleidiCV to match.

Reply:
OK, let's do like this so that HAL should return offset.
I'll prepare PR to OpenCV which documents such behaviour.